### PR TITLE
Dependency bump 2022-04-19

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

**Review by MarcFontaine — APPROVED** [review]
LGTM

**Review by nc6 — APPROVED** [review]

File: cardano-api/gen/Gen/Cardano/Api.hs [hunk 1]
```diff
@@ -83,19 +87,23 @@ genExUnits = do
...
-genCostModel :: Range Int -> Gen Text -> Gen Integer -> Gen Alonzo.CostModel
-genCostModel r gt gi = do
-  map' <- Gen.map r ((,) <$> gt <*> gi)
-  return $ Alonzo.CostModel map'
+genCostModels :: Gen Alonzo.CostModels
+genCostModels = do
+  CostModel cModel <- genCostModel
+  lang <- genLanguage
+  case Alonzo.mkCostModel lang cModel of
+    Left err -> panic . Text.pack $ "genCostModels: " <> err
```
Comment (newhoggy) [inline cardano-api/gen/Gen/Cardano/Api.hs:95]: Are we sure this case never happens?

File: cardano-api/src/Cardano/Api/TxBody.hs [hunk 2]
```diff
@@ -2244,12 +2244,13 @@ fromLedgerTxTotalCollateral era txbody =
   case totalAndReturnCollateralSupportedInEra $ shelleyBasedToCardanoEra era of
     Nothing -> TxTotalCollateralNone
     Just supp ->
-      let totColl = obtainTotalCollateralHasFieldConstraint supp $ getField @"totalCollateral" txbody
-      in TxTotalCollateral supp $ fromShelleyLovelace totColl
+      case obtainTotalCollateralHasFieldConstraint supp $ getField @"totalCollateral" txbody of
+        SNothing ->TxTotalCollateralNone
```
Comment (newhoggy) [inline cardano-api/src/Cardano/Api/TxBody.hs:2248]: Need space after `->`

**Review by JaredCorduan — APPROVED** [review]
LGTM!

**Review by newhoggy — APPROVED** [review]

File: cardano-api/gen/Gen/Cardano/Api.hs (hunk 1 above)
Comment (Jimbo4350) [reply cardano-api/gen/Gen/Cardano/Api.hs:95]: We're not, it depends on `mkEvaluationContext` but: https://input-output-rnd.slack.com/archives/C21UF2WVC/p1649876509637199

It currently returns nothing so for now we just panic. When the plutus team gives more debugging info we can improve.

**Pull request merged**